provider/pkg/provider/provider.go

```diff
@@ -519,6 +520,14 @@ func (p *cfnProvider) Configure(ctx context.Context, req *pulumirpc.ConfigureReq
 		metadata.CfnCustomResourceToken: resources.NewCfnCustomResource(p.name, s3Client, lambdaClient),
 	}
 
+	if autoNaming, ok := vars["aws-native:config:autoNaming"]; ok {
+		var autoNamingConfig autonaming.AutoNamingConfig
```

Comment:
Can we test the unmarshalling? It seems to be untested right now

Reply:
Updated!